docs/DeveloperGuide.adoc

```diff
@@ -153,98 +173,236 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
-// tag::undoredo[]
-=== [Proposed] Undo/Redo feature
-==== Proposed Implementation
+// tag::dueSoon[]
+=== Due Soon Tasks Feature
 
-The undo/redo mechanism is facilitated by `VersionedAddressBook`.
-It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`.
-Additionally, it implements the following operations:
+==== Implementation
 
-* `VersionedAddressBook#commit()` -- Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` -- Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` -- Restores a previously undone address book state from its history.
+The Due Soon Tasks feature allows users to view the tasks that have deadlines within the next week.
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+This feature is implemented using a panel on the main window with a list of tasks due soon that is updated with every "add", "delete" or "refresh".
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+1. Add --
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+   Command: add
```

Comment:
Should add the syntax of the command instead of just the command word itself, e.g.: `add n/ NAME t/ [TAG]...`